src/connectedk8s/azext_connectedk8s/_utils.py

```diff
@@ -379,6 +399,87 @@ def is_guid(guid):
         return False
 
 
+def get_latest_extension_version(extension_name='connectedk8s'):
+    try:
+        import re
+        git_url = "https://raw.githubusercontent.com/Azure/azure-cli-extensions/master/src/{}/setup.py".format(extension_name)
+        response = requests.get(git_url, timeout=10)
+        if response.status_code != 200:
+            logger.info("Failed to fetch the latest version from '%s' with status code '%s' and reason '%s'",
+                        git_url, response.status_code, response.reason)
+            return None
+        for line in response.iter_lines():
+            txt = line.decode('utf-8', errors='ignore')
+            if txt.startswith('VERSION'):
+                match = re.search(r'VERSION = \'(.*)\'$', txt)
+                if match:
+                    return match.group(1)
+                else:
+                    match = re.search(r'VERSION = \"(.*)\"$', txt)
+                    if match:
+                        return match.group(1)
+        return None
+    except Exception as ex:  # pylint: disable=broad-except
+        logger.info("Failed to get the latest version from '%s'. %s", git_url, str(ex))
+        return None
+
+
+def get_existing_extension_version(extension_name='connectedk8s'):
+    from azure.cli.core.extension import get_extensions
+    extensions = get_extensions()
+    if extensions:
+        for ext in extensions:
+            if ext.name == extension_name:
+                return ext.version or 'Unknown'
+
+    return 'NotFound'
+
+
+def check_connectivity(url='https://azure.microsoft.com', max_retries=5, timeout=1):
```

Comment:
Why are we checking connectivity to azure.microsoft.com.
Shouldn't this be azure resource manager endpoint(management.azure.com) ?
Also this value changes based on cloud the customer has set. We first need to check the current cloud and accordingly check connectivity to cloud specific endpoint

Reply:
Another point is that also reach out to the graph endpont(also cloud specific) for fetching the custom locations oid. Probably that connection can also be verified

Reply:
This is general internet connectivity.

Reply:
https://github.com/Azure/azure-cli/blob/b24817537d6f589439ac19178d0ab7672e420598/src/azure-cli-core/azure/cli/core/util.py#L786

Reply:
okay, but what value is it adding to connectedk8s troubleshoot command then? The customer will already get this error in connectedk8s connect

Reply:
This is in troubleshoot command, we can't do some of the things we will be doing in troubleshoot command. If general internet connectivity itself is broken, we are capturing that in logs.